src/aiperf/common/config/user_config.py

```diff
@@ -398,3 +398,66 @@ def validate_concurrency_limits(self) -> Self:
             )
 
         return self
+
+    @model_validator(mode="after")
+    def validate_rankings_token_options(self) -> Self:
+        """Validate rankings token options usage."""
+
+        # Check if prompt input tokens have been changed from defaults
+        prompt_tokens_modified = any(
+            field in self.input.prompt.input_tokens.model_fields_set
+            for field in ["mean", "stddev"]
+        )
+
+        # Check if any rankings-specific token options have been changed from defaults
+        rankings_token_fields = [
+            "rankings_passages_prompt_token_mean",
+            "rankings_passages_prompt_token_stddev",
+            "rankings_query_prompt_token_mean",
+            "rankings_query_prompt_token_stddev",
+        ]
+        rankings_tokens_modified = any(
+            field in self.input.model_fields_set for field in rankings_token_fields
+        )
+
+        # Check if any rankings-specific passage options have been changed from defaults
+        rankings_passages_fields = [
+            "rankings_passages_mean",
+            "rankings_passages_stddev",
+        ]
+        rankings_passages_modified = any(
+            field in self.input.model_fields_set for field in rankings_passages_fields
+        )
+
+        rankings_options_modified = (
+            rankings_tokens_modified or rankings_passages_modified
+        )
+
+        endpoint_type_is_rankings = self.endpoint.type in {
+            "cohere_rankings",
+            "hf_tei_rankings",
+            "nim_rankings",
+        }
+
+        # Validate that rankings options are only used with rankings endpoints
+        if rankings_options_modified and not endpoint_type_is_rankings:
+            raise ValueError(
+                "Rankings-specific options (--rankings-passages-mean, --rankings-passages-stddev, "
+                "--rankings-passages-prompt-token-mean, --rankings-passages-prompt-token-stddev, "
+                "--rankings-query-prompt-token-mean, --rankings-query-prompt-token-stddev) "
+                "can only be used with rankings endpoint types "
+                "(cohere_rankings, hf_tei_rankings, nim_rankings)."
+            )
+
+        # Validate that prompt tokens and rankings tokens are not both set
+        if prompt_tokens_modified and (
+            rankings_tokens_modified or endpoint_type_is_rankings
+        ):
+            raise ValueError(
+                "The --prompt-input-tokens-mean/--prompt-input-tokens-stddev options "
+                "cannot be used together with rankings-specific token options "
```

Comment:
i think it should mention its not allowed for rankings endpoints either, as you are checking for that specifically.

Reply:
added